### PR TITLE
Add checkfile for ScalaCheck

### LINKED ISSUE #20374: Nightly Dotty workflow of 2024-05-10 failed
See https://github.com/scala/scala3/actions/runs/9026620028

### PR DESCRIPTION
**Pull request by hamzaremmal**
[test_scala2_library_tasty]

Closes #20374

### CODE REVIEW

File: tests/init-global/warn/ScalaCheck.check [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- Warning: tests/init-global/warn/ScalaCheck.scala:16:9 ---------------------------------------------------------------
+16 |  object OptMinSuccess extends IntOpt: // warn
+   |         ^
+   |       Cyclic initialization: object OptMinSuccess -> object FirstParser -> object OptMinSuccess. Calling trace:
+   |       ├── object OptMinSuccess extends IntOpt: // warn	[ ScalaCheck.scala:16 ]
+   |       │   ^
+   |       ├── object FirstParser extends CmdLineParser:	[ ScalaCheck.scala:15 ]
+   |       │   ^
+   |       └── val opts = List(OptMinSuccess)	[ ScalaCheck.scala:21 ]
+   |                           ^^^^^^^^^^^^^
```
Comment (liufengyun) [inline tests/init-global/warn/ScalaCheck.check:10]: Sorry that I introduced the problem. As a quick fix, you can change `List` to `Some` in the original test.

Comment (hamzaremmal) [reply tests/init-global/warn/ScalaCheck.check:10]: With this change, it is indeed working locally with the scala2_library_tasty

**Review by liufengyun — APPROVED** [review]
Thank you for taking care of it @hamzaremmal !

**Comment by hamzaremmal** [comment]
> Thank you for taking care of it @hamzaremmal !

No worries, but I would very much suggest opening a ticket to track the original bug

**Comment by liufengyun** [comment]
> No worries, but I would very much suggest opening a ticket to track the original bug

It's an issue well known to us, tracked by https://github.com/scala/scala3/issues/18882. We are actively working on it.

**Comment by hamzaremmal** [comment]
> It's an issue well known to us, tracked by #18882. We are actively working on it.

Oh, I missed this ticket. Good luck !

**Comment by hamzaremmal** [comment]
@liufengyun the CI failed again eventhough it worked for me locally. I will disable the test for now and you can always enbale it later when the fix is ready

**Pull request merged**